### PR TITLE
Answer exercises and point system

### PR DESCRIPTION
**Pull request by Mafusn**
## Description
This pull request includes the implementation of the following PBIs:
[App Course] [Backend] As a waste picker, I want to be able to answer exercises
[App Course] Implement points in Database

## Changes

- UserModel changed
- One route for completing exercises added to userRoutes
- Route for giving points to user
- Level system implemented

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

### CODE REVIEW

File: models/Users.js
Comment (Jatewo) [inline models/Users.js:1]: Looks like parts of this file is using double indentation (2x 2 space tab). Changes are not needed right now. We will have the Linting check up and running by the next sprint, so this should be fixed by then 👍

**Review by Jatewo — CHANGES_REQUESTED** [review]
### Very nice work
Although I haven't gone through the tests line by line it seems well-tested and in general the code seems fine.

I have two overall things which I'd like fixed before approving (one of which is very minor):

1. The response code for an email not found upon login should not be 404 (see my comment on this)
2. A little rework is needed to adhere to the Microsoft convention in the courses/sections routes (see my comment on this)

Please take a look at the other comments as well. Great job! 😄

File: routes/courseRoutes.js
[the remark this answers is not in the document]
Comment (Jatewo) [reply routes/courseRoutes.js:1]: I guess a deletion works 🤣

File: routes/userRoutes.js [hunk 1]
```diff
@@ -52,70 +55,67 @@ router.patch('/:id', requireLogin, async (req, res) => {
     }
 
     if (validFields) {
-      const updatedUser = await User.findByIdAndUpdate( id, { $set: updateFields, modifiedAt: Date.now() }, { new: true })
+      // Extracts the points and level fields from updateFields
```
Comment (JackSMunn) [inline routes/userRoutes.js:58]: This is something we need to look at before we can merge this branch. The current implementation on dev is not passing tests correctly with this implementation. We have created a fix for it, that needs implementing. (let us talk about it tomorrow. 18.10/2023

**Review by JackSMunn — CHANGES_REQUESTED** [review]
See comments. One breaking change that we have created a fix for, lets talk about it.

**Review by JackSMunn — APPROVED** [review]
Approved as my issues are addressed.

**Review by Jatewo — APPROVED** [review]

**Pull request merged**